src/sentry/api/endpoints/project_dynamic_sampling.py

```diff
@@ -281,37 +341,108 @@ def get(self, request: Request, project) -> Response:
+
+            # This condition covers all 3 cases (A, C, D):
+            if root_projects_count == 1 and requested_project:
+                # build parent_project_breakdown
+                parent_project_breakdown = [
+                    {"project_id": project.id, "project": project.slug, "percentege": 1}
+                ]
+                project_breakdown = [
+                    {
+                        "project_id": _project["project_id"],
+                        "project": _project["project"],
+                        "count()": _project["count"],
+                    }
+                    for _project in projects_counts
+                ]
+            # Case B: the most tricky one and requires extra query
+            elif root_projects_count > 1 and requested_project["root"] > 0:
+                parent_project_breakdown = []
+                total_count = sum(
+                    _project["count"] for _project in projects_counts if _project["root"] > 0
+                )
+                for _project in projects_counts:
+                    if _project["root"] > 0:
+                        parent_project_breakdown.append(
+                            {
+                                "project": _project["project"],
+                                "project_id": _project["project_id"],
+                                "percentege": _project["root"] / total_count
+                                if total_count > 0
+                                else 0,
+                            }
+                        )
+                # loop over trace ids, and filter down trace ids where project is sentry
+                parent_trace_ids = [
+                    transaction.get("trace")
+                    for transaction in transactions
+                    if transaction.get("count_root", 0) == 1
+                    and transaction.get("project_id", 0) == project.id
+                ]
+                # and !has:trace.parent_span → project_breakdown attribute.
+                project_breakdown = self.__run_discover_query(
+                    columns=[
+                        "project_id",
+                        "project",
+                        "count()",
+                    ],
+                    query=f"event.type:transaction !has:trace.parent_span trace:[{','.join(parent_trace_ids)}]",
+                    params={
+                        "start": query_time_range.start_time,
+                        "end": query_time_range.end_time,
+                        "project_id": list(projects_in_org),
+                        "organization_id": project.organization.id,
+                    },
+                    limit=20,
+                    referrer="dynamic-sampling.distribution.fetch-project-breakdown",
+                )["data"]
+            # no data
+            else:
+                project_breakdown = []
+                parent_project_breakdown = []
+
+            # Get project breakdown
```

Comment:
After this block of code, you have the `parent_project_breakdown`, then you need to check if the requested project is in the `parent_project_breakdown` 
If it is not, then `project_breakdown = []`
If it is, then filter down trace_ids of that requested_project where root > 0, and then run the query to calculate project breakdown